File: server/app/models/warning.py
from dataclasses import dataclass


@dataclass
class Warning:
    source: str        # "eva1", "eva2", "rover"
    field: str
    value: float
    min: float | None
    max: float | None
    nominal: float | None
    out_of_range: bool
    off_nominal: bool  # in range but deviating from nominal

File: client/app/types/warning.ts
export interface Warning {
    source: string; // "eva1", "eva2", "rover"
    field: string;
    value: number;
    min: number | null;
    max: number | null;
    nominal: number | null;
    out_of_range: boolean;
    off_nominal: boolean; // in range but deviating from nominal
}
